### PR TITLE
book: Add platform support information and tier policy

### PR DESCRIPTION
**Pull request by str4d**
Adapted from the following Rust documents:
- [Platform Support](https://doc.rust-lang.org/stable/rustc/platform-support.html)
- [Target Tier Policy](https://doc.rust-lang.org/stable/rustc/target-tier-policy.html)

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
ACK

File: doc/book/src/dev/platform-tier-policy.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+The availability or tier of a platform in Zcash releases is not a hard stability guarantee
+about the future availability or tier of that platform. Higher-level platform tiers are an
+increasing commitment to the support of a platform, and we will take that commitment and
+potential disruptions into account when evaluating the potential demotion or removal of a
+platform that has been part of a stable release. The promotion or demotion of a platform
+will not generally affect existing stable releases, only current development and future
+releases.
+
+In this policy, the words "must" and "must not" specify absolute requirements that a
```
Comment (daira) [inline doc/book/src/dev/platform-tier-policy.md:50]: Use capitalised keywords.

File: doc/book/src/dev/platform-tier-policy.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  for the platform, using cross-compilation if possible. If the platform supports running
+  binaries, or running tests (even if they do not pass), the documentation must explain
+  how to run such binaries or tests for the platform, using emulation if possible or
+  dedicated hardware if necessary.
+- Tier 3 platforms must not impose burden on the authors of pull requests, or other
+  developers in the community, to maintain the platform. In particular, do not post
+  comments (automated or manual) on a PR that derail or suggest a block on the PR based on
+  a tier 3 platform. Do not send automated messages or notifications (via any medium,
+  including via @) to a PR author or others involved with a PR regarding a tier 3
+  platform, unless they have opted into such messages.
```
Comment (daira) [inline doc/book/src/dev/platform-tier-policy.md:75]: I don't see why reviewers should avoid commenting on portability to tier 3 platforms; that seems like a reasonable kind of review comment to me. It's not the same as blocking on support for a tier 3 platform.

File: doc/book/src/user/platform-support.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Tier 3 platforms are those for which the `zcashd` codebase has support, but ECC does not
+require builds or tests to pass, so these may or may not work. Official builds are not
+available.
+
+| target                  | OS           | notes |
+| ----------------------- | ------------ | ----- |
+| `x86_64-pc-linux-gnu`   | Arch         |
+| `x86_64-w64-mingw32`    | Windows      | 64-bit MinGW |
+| `x86_64-apple-darwin16` | macOS 10.14+ |
+| `aarch64-linux-gnu`     | ARM64 Linux  |
```
Comment (daira) [inline doc/book/src/user/platform-support.md:50]: It's there still any OpenBSD support in the build system? I remember that used to work.

**Review by daira — APPROVED** [review]
ACK with non-blocking comments.

File: doc/book/src/user/platform-support.md (hunk 3 above)
Comment (str4d) [reply doc/book/src/user/platform-support.md:50]: Ah yes, indeed there is. I populated this table from the CI workers, and we don't have one for this. Which makes it the perfect tier 3 example!

File: doc/book/src/dev/platform-tier-policy.md (hunk 1 above)
Comment (str4d) [reply doc/book/src/dev/platform-tier-policy.md:50]: Hmm, RFC 2119 is defined in the context of a specification, but this is a policy.

**Comment by mdr0id** [comment]
utACK

File: doc/book/src/dev/platform-tier-policy.md (hunk 2 above)
Comment (daira) [reply doc/book/src/dev/platform-tier-policy.md:75]: Commenting at all on a PR notifies the PR author, though. And if you consider a PR author to have opted into notifications, then the "don't notify the PR author" part of this becomes literally a no-op. I have no objection to saying that the PR shouldn't be blocked, that's not what my comment was about.

Comment (str4d) [reply doc/book/src/dev/platform-tier-policy.md:75]: It covers everything other than PR authors in their own PRs though (e.g. other issues, including the one that a PR might be addressing), so without any suggested alternative wording, I'm content to leave it as more restrictive by default, and then handle any issues / relaxations as they arise.

**Pull request merged**